crates/astria-sequencer/src/app/mod.rs

```diff
@@ -942,7 +1002,7 @@ impl App {
         )
         .wrap_err("failed to convert block info and data to SequencerBlock")?;
         state_tx
-            .put_sequencer_block(sequencer_block)
+            .put_sequencer_block(sequencer_block.clone())
```

Comment:
this may or may not be a bad idea, since we are cloning a potentially big struct which might make a db read more efficient?